frontend/src/pages/Battle/Battle.tsx

```diff
@@ -11,11 +11,52 @@ import { Button } from "components/Button";
 import { useUnit } from "effector-react";
 import { logsStore } from "model/logs";
 import { battle } from "model/battleLogs";
-// import { userStore } from "model/user";
-// import { isEmpty } from "lodash";
-// import { useAlert } from "@gear-js/react-hooks";
 
 export type BattleProps = {};
+
+type TurnTypes = "move" | "attack" | "miss" | "rest";
+type TurnGeneric<K extends TurnTypes, T> = { [key in K]: T };
+type Turn =
+  | TurnGeneric<"move", { position: number }>
+  | TurnGeneric<"attack", { damage: number; position: number }>
+  | TurnGeneric<"miss", { position: number }>
+  | TurnGeneric<"rest", { position: number }>;
+
+type BattleLogs = {
+  logs: Array<{
+    c1: string;
+    c2: string;
+    turns: Turn[];
+    winner: string;
+  }>;
+  winner: string;
+};
+
+const actions = {
```

Comment:
think u should create a new file textMap.ts for this

Reply:
done